### ISSUE TITLE
INTERNAL: Panic! This is a bug!

### ISSUE THREAD

**Issue by nwbirnie** [grpc/grpc-java#6590]
### What version of gRPC-Java are you using?

1.21.0

### What is your environment?

Long running server process. Attempting to make a call after some time idle, the reschedule() call fails with Panic.

### What did you expect to see?

API call succeeds.

### What did you see instead?

Panic.

### Steps to reproduce the bug

Please see related [issue on Google Ads Java client library](https://github.com/googleads/google-ads-java/issues/194).

**Comment by nwbirnie** [comment]
Related stacktrace:

```
[Channel<1702603>: (googleads.googleapis.com:443)] Uncaught exception in the SynchronizationContext. Panic! 
java.util.concurrent.RejectedExecutionException: event executor terminated
  at io.grpc.netty.shaded.io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:845)
  at io.grpc.netty.shaded.io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:328)
  at io.grpc.netty.shaded.io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:321)
  at io.grpc.netty.shaded.io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:756)
  at io.grpc.netty.shaded.io.netty.util.concurrent.AbstractScheduledEventExecutor.schedule(AbstractScheduledEventExecutor.java:232)
  at io.grpc.netty.shaded.io.netty.util.concurrent.AbstractScheduledEventExecutor.schedule(AbstractScheduledEventExecutor.java:155)
  at io.grpc.netty.shaded.io.netty.util.concurrent.AbstractEventExecutorGroup.schedule(AbstractEventExecutorGroup.java:50)
  at io.grpc.netty.shaded.io.netty.util.concurrent.AbstractEventExecutorGroup.schedule(AbstractEventExecutorGroup.java:32)
  at io.grpc.internal.Rescheduler.reschedule(Rescheduler.java:63)
  at io.grpc.internal.ManagedChannelImpl.rescheduleIdleTimer(ManagedChannelImpl.java:408)
  at io.grpc.internal.ManagedChannelImpl.exitIdleMode(ManagedChannelImpl.java:366)
  at io.grpc.internal.ManagedChannelImpl$ChannelTransportProvider$1ExitIdleModeForTransport.run(ManagedChannelImpl.java:467)
  at io.grpc.SynchronizationContext.drain(SynchronizationContext.java:88)
  at io.grpc.SynchronizationContext.execute(SynchronizationContext.java:120)
  at io.grpc.internal.ManagedChannelImpl$ChannelTransportProvider.get(ManagedChannelImpl.java:471)
  at io.grpc.internal.ClientCallImpl.startInternal(ClientCallImpl.java:255)
  at io.grpc.internal.ClientCallImpl.start(ClientCallImpl.java:188)
  at io.grpc.internal.CensusTracingModule$TracingClientInterceptor$1.start(CensusTracingModule.java:394)
  at io.grpc.internal.CensusStatsModule$StatsClientInterceptor$1.start(CensusStatsModule.java:695)
  at io.grpc.ForwardingClientCall.start(ForwardingClientCall.java:32)
  at com.google.api.gax.grpc.GrpcHeaderInterceptor$1.start(GrpcHeaderInterceptor.java:94)
  at io.grpc.ForwardingClientCall.start(ForwardingClientCall.java:32)
  at com.google.ads.googleads.lib.logging.LoggingInterceptor$1.start(LoggingInterceptor.java:78)
  at io.grpc.stub.ClientCalls.startCall(ClientCalls.java:310)
  at io.grpc.stub.ClientCalls.asyncUnaryRequestCall(ClientCalls.java:282)
  at io.grpc.stub.ClientCalls.futureUnaryCall(ClientCalls.java:191)
  at com.google.api.gax.grpc.GrpcDirectCallable.futureCall(GrpcDirectCallable.java:58)
  at com.google.api.gax.grpc.GrpcUnaryRequestParamCallable.futureCall(GrpcUnaryRequestParamCallable.java:65)
  at com.google.api.gax.grpc.GrpcExceptionCallable.futureCall(GrpcExceptionCallable.java:64)
  at com.google.api.gax.rpc.AttemptCallable.call(AttemptCallable.java:86)
  at com.google.api.gax.rpc.RetryingCallable.futureCall(RetryingCallable.java:63)
  at com.google.api.gax.rpc.RetryingCallable.futureCall(RetryingCallable.java:41)
  at com.google.ads.googleads.lib.ExceptionTransformingCallable.futureCall(ExceptionTransformingCallable.java:51)
  at com.google.api.gax.rpc.PagedCallable.futureCall(PagedCallable.java:62)
  at com.google.api.gax.rpc.UnaryCallable$1.futureCall(UnaryCallable.java:126)
  at com.google.api.gax.rpc.UnaryCallable.futureCall(UnaryCallable.java:87)
  at com.google.api.gax.rpc.UnaryCallable.call(UnaryCallable.java:112)
  at com.google.ads.googleads.v2.services.GoogleAdsServiceClient.search(GoogleAdsServiceClient.java:247)
  at com.expedia.www.mtt.googleads.service.gha.client.BaseGhaReadOnlyRepo$trySearch$1.invoke(GhaRepo.kt:50)
 .......
  at com.expedia.www.mtt.googleads.service.resources.GoogleAdsUploadResource.process(GoogleAdsUploadResource.kt:54)
  at sun.reflect.GeneratedMethodAccessor468.invoke(Unknown Source)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:498)
  at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81)
  at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:144)
  at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:161)
  at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:160)
  at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:99)
  at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:389)
  at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:347)
  at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:102)
  at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:326)
  at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
  at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
  at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
  at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
  at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
  at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317)
  at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305)
  at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154)
  at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:473)
  at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:427)
  at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:388)
  at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:341)
  at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:228)
  at io.dropwizard.jetty.NonblockingServletHolder.handle(NonblockingServletHolder.java:49)
  at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1650)
  at io.dropwizard.servlets.ThreadNameFilter.doFilter(ThreadNameFilter.java:35)
  at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
  at io.dropwizard.jersey.filter.AllowedMethodsFilter.handle(AllowedMethodsFilter.java:45)
  at io.dropwizard.jersey.filter.AllowedMethodsFilter.doFilter(AllowedMethodsFilter.java:39)
  at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
  at ch.qos.logback.access.servlet.TeeFilter.doFilter(TeeFilter.java:53)
  at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
  at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
  at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188)
  at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
  at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168)
  at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
  at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166)
  at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
  at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
  at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
  at com.codahale.metrics.jetty9.InstrumentedHandler.handle(InstrumentedHandler.java:239)
  at io.dropwizard.jetty.RoutingHandler.handle(RoutingHandler.java:52)
  at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:455)
  at io.dropwizard.jetty.BiDiGzipHandler.handle(BiDiGzipHandler.java:63)
  at org.eclipse.jetty.server.handler.RequestLogHandler.handle(RequestLogHandler.java:56)
  at org.eclipse.jetty.server.handler.StatisticsHandler.handle(StatisticsHandler.java:169)
  at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
  at org.eclipse.jetty.server.Server.handle(Server.java:530)
  at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:347)
  at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:256)
  at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279)
  at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:102)
  at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124)
  at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:247)
  at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:140)
  at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
  at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:382)
  at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:708)
  at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:626)
  at java.lang.Thread.run(Thread.java:748)
```

**Comment by voidzcy** [comment]
Based on the [stacktrace posted here](https://github.com/googleads/google-ads-java/issues/194#issue-529528056) (`io.grpc.internal.SharedResourceHolder$2@1b39c0af
io.grpc.netty.shaded.io.netty.channel.ChannelException: eventfd_write() failed: Bad file descriptor`), this looks to be the same issue as #6002, which has been fixed since [1.24.0](https://github.com/grpc/grpc-java/releases/tag/v1.24.0). Please try with newer versions to see if your issue goes away.

**Comment by dapengzhang0** [comment]
Based on the fix of https://github.com/googleads/google-ads-java/issues/194, we can conclude that it's the same issue as #6002.
